clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1272,10 +1281,19 @@ public void handle(OffsetFetchResponse response, RequestFuture<Map<TopicPartitio
         private final Sensor assignCallbackSensor;
         private final Sensor loseCallbackSensor;
         private final Sensor pollInterval;
+        private final Sensor metadataRequestRateSensor;
 
         private ConsumerCoordinatorMetrics(Metrics metrics, String metricGrpPrefix) {
             this.metricGrpName = metricGrpPrefix + "-coordinator-metrics";
 
+            this.metadataRequestRateSensor = metrics.sensor("consumer-coordinator-metadata-request-rate");
```

Comment:
I doubt the users could care the difference between the MD requests triggered by the KafkaConsumer and those triggered by the ConsumerCoordinator.
Since they both share the same ConsumerMetadata object, do you think we can move the sensor to be inside the ConsumerMetadata class?

Reply:
hi @gitlw , there are downstream function calls that pass the ConsumerMetadata object into Consumer client directly, which does not include  client specific metric config / metric reporters in this case. This would raise inconsistency for the sensor recorded values (default empty metrics vs. metrics passed into ConsumerMetadata constructor).